src/main/java/com/zendesk/maxwell/monitoring/MaxwellMetrics.java

```diff
@@ -123,16 +118,21 @@ public String metricName(String... names) {
 
 	@Override
 	public MetricRegistry getRegistry() {
-		return metricRegistry;
+		return config.metricRegistry;
 	}
 
 	public void startBackgroundTasks(MaxwellContext context) throws IOException {
-		String metricsReportingType = config.metricsReportingType;
-		if (metricsReportingType != null && metricsReportingType.contains(reportingTypeHttp)) {
-			LOGGER.info("Metrics http server starting");
-			new MaxwellHTTPServer(config.metricsHTTPPort, metricRegistry, healthCheckRegistry, context);
-			healthCheckRegistry.register("MaxwellHealth", new MaxwellHealthCheck(context.getProducer()));
-			LOGGER.info("Metrics http server started on port " + config.metricsHTTPPort);
+		context.getConfig().healthCheckRegistry.register("MaxwellHealth", new MaxwellHealthCheck(context.getProducer()));
+		MaxwellHTTPServer.startIfRequired(context);
```

Comment:
This feels misplaced now, since we may be starting the HTTP server even if HTTP metrics are disabled. Perhaps move this `startIfRequired` call up to MaxwellContext's `start()` since it might be enabled by both metrics and diagnostics?

Reply:
Good catch, I promise I had this thought, until I forgot to do it -_-||